src/libs/ajax/Metrics.ts

```diff
@@ -6,21 +6,21 @@ import { getConfig } from 'src/libs/config';
 import { withErrorIgnoring } from 'src/libs/error';
 import { MetricsEventName } from 'src/libs/events';
 import * as Nav from 'src/libs/nav';
-import { authStore, getSessionId } from 'src/libs/state';
+import { AuthState, authStore, getSessionId } from 'src/libs/state';
 import { v4 as uuid } from 'uuid';
 
 export const Metrics = (signal?: AbortSignal) => {
   const captureEventFn = async (event: MetricsEventName, details: Record<string, any> = {}): Promise<void> => {
     await ensureAuthSettled();
-    const { signInStatus, registrationStatus } = authStore.get(); // NOTE: This is intentionally read after ensureAuthSettled
-    const isRegistered = signInStatus === 'signedIn' && registrationStatus === 'registered';
-    if (!isRegistered) {
-      authStore.update(
-        _.update('anonymousId', (id) => {
-          return id || uuid();
-        })
-      );
+    const state: AuthState = authStore.get();
+    const isRegistered = state.registrationStatus === 'registered';
+    if (state.anonymousId === undefined) {
```

Comment:
Adding `isRegistered` to the condition might make this clearer.

Also probably worth a comment, given how long we spent trying to figure out exactly how this worked.

```suggestion
    // If a user has not logged in, or has logged in but has not registered, ensure an anonymous ID has been generated.
    // The anonymous ID is used to associate events triggered by the anonymous user.
    // If the anonymous user registers during this session, the anonymous ID will be linked to their actual user ID.
    if (!isRegistered && state.anonymousId === undefined) {
```